### PR TITLE
Fix enable LLVM for JIT compilation in CMake

### PR DESCRIPTION
**Pull request by kitaisreal**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable build with JIT compilation by default.

### CODE REVIEW

File: contrib/llvm-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -2,11 +2,12 @@
 # tools to be build for host architecture and everything else for target architecture (e.g. AArch64)
 # Possible workaround is to use llvm-tblgen from some package...
 # But lets just enable LLVM for native builds
-if (CMAKE_CROSSCOMPILING OR SANITIZE STREQUAL "undefined")
+if (NOT (${CMAKE_SYSTEM_NAME} MATCHES ${CMAKE_HOST_SYSTEM_NAME}) OR SANITIZE STREQUAL "undefined")
```
Comment (kitaisreal) [inline contrib/llvm-cmake/CMakeLists.txt:5]: CMAKE_CROSSCOMPILING can be true even without cross compilation
```
This variable will be set to true by CMake if the [CMAKE_SYSTEM_NAME]
(https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html#variable:CMAKE_SYSTEM_NAME) 
variable has been set manually (i.e. in a toolchain file or as a cache entry from the [cmake]
```
https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html

**Comment by kitaisreal** [comment]
@Mergifyio update

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by Felixoid** [comment]
Yep, we can release a new version

**Comment by kitaisreal** [comment]
@Mergifyio update

**Comment by tavplubix** [comment]
@kitaisreal you forgot to take a look at failed tests before merging again:
Stateless tests (release, DatabaseReplicated, actions) - `02152_http_external_tables_memory_tracking`
Stateless tests (release, actions) - `02152_http_external_tables_memory_tracking`
Stateless tests (release, s3 storage, actions) - `02152_http_external_tables_memory_tracking`
Stateless tests (release, wide parts enabled, actions) - `02152_http_external_tables_memory_tracking`

This test was not flaky for last 3 weeks: http://a137ed393136a448a8a0b796e91304c4-2000240599.us-east-1.elb.amazonaws.com:8123/play?user=default#c2VsZWN0IAp0b1N0YXJ0T2ZEYXkoY2hlY2tfc3RhcnRfdGltZSkgYXMgZCwKY291bnQoKSwgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikKZnJvbSBgZ2gtZGF0YWAuY2hlY2tzIHdoZXJlICcyMDIyLTAxLTAxJyA8PSBjaGVja19zdGFydF90aW1lIGFuZCB0ZXN0X25hbWU9JzAyMTUyX2h0dHBfZXh0ZXJuYWxfdGFibGVzX21lbW9yeV90cmFja2luZycgYW5kIHRlc3Rfc3RhdHVzIGluICgnRkFJTCcsICdGTEFLWScpIGdyb3VwIGJ5IGQgb3JkZXIgYnkgZA==

So seems like it was broken here somehow. I will try to revert.

**Comment by kitaisreal** [comment]
@tavplubix I take a look this change just revert other pull request that breaks LLVM compilation, everything must work as before.

**Comment by azat** [comment]
>Stateless tests (release, actions) - 02152_http_external_tables_memory_tracking

The problem was in settings randomization, in particular `max_block_size=67990 ` for `02152_http_external_tables_memory_tracking`, fixed in #35745

**Pull request merged**